client/core/core.go

```diff
@@ -1677,7 +1695,7 @@ func (c *Core) NewDepositAddress(assetID uint32) (string, error) {
 	}
 
 	c.walletMtx.Lock()
-	w.address = addr
+	w.setAddress(addr)
 	c.walletMtx.Unlock()
```

Comment:
I think this locking `Core.walletMtx` was a bug. `walletMtx` is meant to guard the `wallets` map.  The `xcWallet` has it's own mutex, which `setAddress` uses.

Review the use of `walletMtx` and the declaration of the mutex and the `wallets` field in the `Core` struct to see what I mean.  @buck54321 please confirm that there was not another consideration here, but it appears to be a misuse of `Core`'s `walletMtx` instead of the `xcWallet` mutex.

Reply:
Oh, that's make sense, will update it, this might explain the deadlock I had when generating new address